nixos/modules/services/networking/pppd.nix

```diff
@@ -0,0 +1,137 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+{
+  meta = {
+    maintainers = with maintainers; [ danderson ];
+  };
+
+  options = {
+    services.pppd = {
+      package = mkOption {
+        default = pkgs.ppp;
+        defaultText = "pkgs.ppp";
+        type = types.package;
+        description = "pppd package to use.";
+      };
+
+      peers = mkOption {
+        default = {};
+        type = types.loaOf (types.submodule (
```

Comment:
`loaOf` should be being deprecated. See https://github.com/NixOS/nixpkgs/pull/63103 . I would use `attrsOf` here but if you might need multiple submodules per name (or if ordering matters) then use list